src/common/crypto/mbedtls/driverwrapper.cpp

```diff
@@ -222,6 +225,24 @@ static int ExportECPublicKeyToDER(
     return 0;
 }
 
+static bool IsKeyInBuiltinList(psa_key_id_t keyID)
+{
+    auto ret = sBuiltinKeys.FindIf([&](const KeyDescription& key) { return key.mKeyID == keyID; });
+
+    return ret.mError.IsNone();
+}
+
+static aos::RetWithError<size_t> FindFreeSlot()
```

Comment:
Why not return iterator instead of calculating index?

Reply:
Done